tests/api/api/marketplace/test_marketplace.py

```diff
@@ -304,3 +304,20 @@ def test_marketplace_catalog_apis(
     function_modified_name = item["metadata"]["name"].replace("_", "-")
 
     assert yaml_function_name == function_modified_name
+
+
+def test_marketplace_get_asset(
+    db: Session, client: TestClient, k8s_secrets_mock: tests.api.conftest.K8sSecretsMock
+) -> None:
+    possible_assets = ["docs", "source", "example", "function"]
+    sources = client.get("marketplace/sources").json()
+    source_name = sources[0]["source"]["metadata"]["name"]
+    catalog = client.get(f"marketplace/sources/{source_name}/items").json()
+    for _ in range(10):
+        item = random.choice(catalog["catalog"])
+        asset_type = random.choice(possible_assets)
+        response = client.get(
+            f"marketplace/sources/{source_name}/items/{item['metadata']['name']}/assets/{asset_type}"
+        )
+        assert response.status_code == http.HTTPStatus.OK.value
+        assert response.content == bytes(response.text, encoding="utf-8")
```

Comment:
you are comparing the response with the response (what if it's empty or is malformed?)
is it possible to get the asset in a different way and compare with the response?